src/steamship/data/tags/tag.py

```diff
@@ -15,17 +15,40 @@ class TagQueryRequest(Request):
 
 
 class Tag(CamelModel):
+    # Steamship client.
     client: Client = Field(None, exclude=True)
+
+    # ID of the tag in the database.
     id: str = None
+
+    # ID of the file associated with the tag.
     file_id: str = None
-    block_id: str = None
+
+    # ID of the block associated with the tag. If not None, `start_idx` and `end_idx` should be set.
+    block_id: Optional[str] = None
+
+    # The kind of tag. See the ``TagKind`` enum class for suggestions.
     kind: str = None  # E.g. ner
-    name: str = None  # E.g. person
-    value: Dict[str, Any] = None  # JSON Metadata
-    start_idx: int = None  # w/r/t block.text. None means 0 if blockId is not None
-    end_idx: int = None  # w/r/t block.text. None means -1 if blockId is not None
+
+    # The name of tag. See the ``DocTag``, ``TokenTag``, etc enum classes for suggestions.
+    name: Optional[str] = None  # E.g. person
+
+    # The value payload of the tag. Always a JSON-style object.
+    value: Optional[Dict[str, Any]] = None
+
+    # Start-index of the text covered by the tag, using python slice semantics.
```

Comment:
I don't know what you meant to convey with 'using python slice semantics', but I think that comment should be removed.  You can't, for example, use negative indices to imply an offset from the back, nor does omitting just start or end mean the beginning/end, so I don't think these use slice semantics at all.

Reply:
I would highlight that we're using a character index (vs for example a token index)

Reply:
Thanks for the catch -- I'll update to mention character index and just be specific about start inclusivity and end exclusivity